src/mono/mono/mini/interp/transform.c

```diff
@@ -4743,6 +4752,14 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 					/* public ByReference(ref T value) */
 					g_assert (csignature->hasthis && csignature->param_count == 1);
 					interp_add_ins (td, MINT_INTRINS_BYREFERENCE_CTOR);
+				} else if (m_class_get_image (klass) == mono_defaults.corlib &&
+						(!strcmp (m_class_get_name (m->klass), "Span`1") ||
+						!strcmp (m_class_get_name (m->klass), "ReadOnlySpan`1")) &&
+						csignature->param_count == 2 &&
+						csignature->params [0]->type == MONO_TYPE_PTR &&
+						!type_has_references (mono_method_get_context (m)->class_inst->type_argv [0])) {
```

Comment:
@vargaz type_has_references is hackish. Do we have some api in the runtime to check if a type has references, or to get the class associated with a type ?

Reply:
mono_class_from_mono_type ().

Reply:
not sure how I couldn't find it since I even used it before in interp